pkg/prometheus/promcfg.go

```diff
@@ -1444,6 +1444,22 @@ func generateAddressShardingRelabelingRules(relabelings []yaml.MapSlice, shards
 	return generateAddressShardingRelabelingRulesWithSourceLabel(relabelings, shards, "__address__")
 }
 
+func (cg *ConfigGenerator) generateAddressShardingRelabelingRulesIfMissing(relabelings []yaml.MapSlice, shards int32) []yaml.MapSlice {
+	alreadyHasShardLabel := false
+	for i, relabeling := range relabelings {
+		for _, relabelItem := range relabeling {
+			if relabelItem.Key == "action" && relabelItem.Value == "hashmod" {
+				level.Debug(cg.logger).Log("msg", "found existing hashmod relabeling rule, skipping", "idx", i)
+				alreadyHasShardLabel = true
+			}
+		}
+	}
+	if !alreadyHasShardLabel {
+		return generateAddressShardingRelabelingRules(relabelings, shards)
+	}
+	return relabelings
```

Comment:
just a suggestion, feel free to ignore
```suggestion
	for i, relabeling := range relabelings {
		for _, relabelItem := range relabeling {
			if relabelItem.Key == "action" && relabelItem.Value == "hashmod" {
				level.Debug(cg.logger).Log("msg", "found existing hashmod relabeling rule, skipping", "idx", i)
				return generateAddressShardingRelabelingRules(relabelings, shards)
			}
		}
	}

	return relabelings
```

Reply:
cleaner :+1: